packages/picasso.js/src/core/chart/index.js

```diff
@@ -176,6 +177,54 @@ import componentCollectionFn from './component-collection';
  * }
  */
 
+/**
+ * @typedef {object} RendererSettings
+ * @property {RendererSettings~TransformFunction} [transform] Setting for applying transform without re-rendering the whole component completely.
+ * @property {RendererSettings~CanvasBufferSize} [canvasBufferSize] Specifies the size of buffer canvas (used together with transform setting).
+ * @experimental
+ */
+
+/**
+ * Should return a transform object if transformation should be applied, otherwise undefined or a falsy value.
+ * Transforms can be applied with the canvas, svg and dom renderer.
+ * !Transform is applied when running chart.update with partialData set to true, see example.
+ * @typedef {function} RendererSettings~TransformFunction
+ * @returns {TransformObject}
+ * @experimental
+ * @example
+ * const pointComponentDef = {
+ *   type: 'point',
+ *   rendererSettings: {
+ *     tranform() {
+ *       if(shouldApplyTransform) {
+ *         return { a: 1, b: 0, c: 0, d: 1, e: x, f: y };
+ *       }
+ *     }
+ *   }
+ *   data: {
+ * // ............
+ *
+ * chart.update({ partialData: true });
+ */
+
+/**
+ * An object containing width and height of the canvas buffer or a function returning an object on that format.
+ * Gets a rect object as input parameter.
+ * @typedef {function|object} RendererSettings~CanvasBufferSize
+ * @experimental
+ */
+
+/**
+ * A format to represent a transformation.
+ * @typedef {object} TransformObject
+ * @property {number} a Horizontal scaling
+ * @property {number} b Horizontal skewing
+ * @property {number} c Vertical skewing
+ * @property {number} d Vertical scaling
+ * @property {number} e Horizontal moving
+ * @property {number} f Vertical moving
```

Comment:
have you considered using more descriptive names

Reply:
Yeah, but it also seems to be some kind of standard around this naming (look in here for instance: https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Transformations#transforms), so I feel it's ok to stick to this...

Reply:
the last two in not that standardized see: https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix()

Reply:
Allright.. I've given the parameters to more descriptive names now.